azblob/sas_service.go

```diff
@@ -30,7 +31,18 @@ type BlobSASSignatureValues struct {
 // the proper SAS query parameters.
 func (v BlobSASSignatureValues) NewSASQueryParameters(sharedKeyCredential *SharedKeyCredential) (SASQueryParameters, error) {
 	resource := "c"
-	if v.BlobName == "" {
+	if v.SnapshotTime.IsZero() {
+		resource = "bs"
+
+		perms := &ContainerSASPermissions{}
+		if err := perms.Parse(v.Permissions); err != nil {
+			return SASQueryParameters{}, err
+		}
+		v.Permissions = perms.String()
+		if v.Version == "" {
```

Comment:
isn't this redundant? line 61

Reply:
Actually you're right, it is redundant, I didn't realize while I was restructuring